### PR TITLE
Use sphinx rst syntax

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

I've found a few minor document issues in recently introduced pages.

## Description of the changes

Fix them and fix a typo in a private method name.

### CODE REVIEW

**Comment by eukaryo** [comment]
@nabenabe0928 Could you review this PR?

**Review by eukaryo — APPROVED** [review]
LGTM. I checked the following docs:
https://optuna--5345.org.readthedocs.build/en/5345/reference/generated/optuna.pruners.WilcoxonPruner.html#optuna.pruners.WilcoxonPruner
https://optuna--5345.org.readthedocs.build/en/5345/reference/samplers/generated/optuna.samplers.GPSampler.html#optuna.samplers.GPSampler

File: optuna/pruners/_wilcoxon.py [hunk 1]
```diff
@@ -29,18 +29,20 @@ class WilcoxonPruner(BasePruner):
...
 
-    When you use this pruner, you must call `Trial.report(value, step)` function for each step (instance id) with
+    When you use this pruner, you must call ``Trial.report(value, step)`` method for each step (instance id) with
     the evaluated value. The instance id may not be in ascending order.
     This is different from other pruners in that the reported value need not converge
-    to the real value. (To use pruners such as `SuccessiveHalvingPruner` in the same setting, you must provide e.g.,
-    the historical average of the evaluated values.)
+    to the real value. To use pruners such as :class:`~optuna.pruners.SuccessiveHalvingPruner`
+    in the same setting, you must provide e.g.,
+    the historical average of the evaluated values.
```
Comment (nabenabe0928) [inline optuna/pruners/_wilcoxon.py:45]:
```suggestion
    in the same setting, you must provide e.g., the historical average of the evaluated values.
```

**Review by nabenabe0928 — COMMENTED** [review]
I left a minor comment, but LGTM!

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

File: optuna/samplers/_gp/sampler.py [hunk 2]
```diff
@@ -54,27 +54,27 @@ class GPSampler(BaseSampler):
         - Quasi-Monte Carlo (QMC) sampling to optimize the acquisition function.
 
     .. note::
-        This sampler requires `scipy` and `pytorch`.
-        You can install these dependencies with `pip install scipy pytorch`.
+        This sampler requires ``scipy`` and ``pytorch``.
+        You can install these dependencies with ``pip install scipy pytorch``.
```
Comment (nabenabe0928) [inline optuna/samplers/_gp/sampler.py:58]:
```suggestion
        This sampler requires ``scipy`` and ``torch``.
        You can install these dependencies with ``pip install scipy torch``.
```

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
I missed one thing!

File: optuna/samplers/_gp/sampler.py (hunk 2 above)
Comment (nzw0301) [reply optuna/samplers/_gp/sampler.py:58]: Nice🙏

**Review by nabenabe0928 — APPROVED** [review]

**Pull request merged**